release/models/aft/openconfig-aft-common.yang

```diff
@@ -244,11 +258,134 @@ submodule openconfig-aft-common {
+            in the list encap-header.
+
+            Each entry in the list must indicate an encapsulation type and
+            populate a container with the parameters for that encapsulation
+            header.";
+          list encap-header {
+            description
+              "A list of headers added on top of a packet.  The first entry
+              in the list is the inner-most packet at the bottom of the
+              stack.
+
+              For example, in an encapsulation stack with MPLS in UDP, the
+              first index in the list is the MPLS packet and the second
+              index is UDP.";
+            key "index";
+
+            leaf index {
+              type leafref {
+                path "../state/index";
+              }
+              description
+                "A unique index identifying an encapsulation header in a stack
+                of encapsulation headers.";
+            }
+
+            container state {
+              description
+                "State parameters relating to encapsulation headers.";
+              uses aft-common-nexthop-encap-headers-state;
+            }
+
+            container gre {
+              when "../state/type = 'oc-aftt:GRE'";
+              description
+                "Container of nodes for GRE encapsulation.";
+              container state {
+                description
+                  "State parameters relating to GRE encapsulation headers.";
+                uses aft-common-entry-nexthop-gre-state;
+              }
+            }
+
+            container ipv4 {
+              when "../state/type = 'oc-aftt:IPV4'";
+              description
+                "Container of nodes for IPv4 encapsulation.";
+              container state {
+                description
+                  "State parameters relating to IP encapsulation headers.";
+                uses aft-common-entry-nexthop-ip-state;
+              }
+            }
+
+            container ipv6 {
+              when "../state/type = 'oc-aftt:IPV6'";
+              description
+                "Container of nodes for IPv6 encapsulation.";
+              container state {
+                description
+                  "State parameters relating to IP encapsulation headers.";
+                uses aft-common-entry-nexthop-ip-state;
+              }
+            }
+
+            container mpls {
+              when "../state/type = 'oc-aftt:MPLS'";
+              description
+                "Container of nodes for MPLS encapsulation.";
+              container state {
+                description
+                  "State parameters relating to MPLS encapsulation headers.";
+                uses aft-common-entry-nexthop-mpls-state;
+              }
+            }
+
+            container udp {
+              when "../state/type = 'oc-aftt:UDPV4' or " +
+              "../state/type = 'oc-aftt:UDPV6'";
+              description
```

Comment:
Can we add something to the description here to say that this implies both an IP *and* a UDP header are applied? Maybe we should also add something to the IPv4 and IPv6 cases to say that this is just a bare IP header.